### PR TITLE
Refactor/rrd to next router

### PR DESCRIPTION
**Pull request by waynewee**
Attempt to remove react router dom, need to refactor all CSR stuff using window variable to completely remove rrd. 

This PR contains only refactoring of useLocation and useHistory

### CODE REVIEW

File: src/AppContainer.tsx [hunk 1]
```diff
@@ -1,28 +1,28 @@
 import { NetworkBar, Overlay } from "@govtechsg/tradetrust-ui-components";
 import React, { useEffect, useState } from "react";
-import { useLocation } from "react-router-dom";
 import { Footer } from "./components/Layout/Footer";
 import { NavigationBar, leftNavItems, rightNavItems } from "./components/Layout/NavigationBar";
 import { NETWORK } from "./config";
 import { Routes, routes } from "./routes";
 import styled from "@emotion/styled";
+import { useRouter } from "next/router";
```
Comment (Nebulis) [inline src/AppContainer.tsx:8]: remove react-router-dom from package.json ?

File: src/components/HomePageContent/HomePageContainer.tsx [hunk 2]
```diff
@@ -36,12 +35,14 @@ export const HomePageContainer = (): React.ReactElement => {
...
-      const anchorStr = decodeURIComponent(location.hash.substr(1));
+    const { query, asPath } = router;
+    if (Object.values(query).length > 0) {
+      const hash = asPath.split("#")[1];
+
+      const anchorStr = decodeURIComponent(hash);
       const anchor = anchorStr ? JSON.parse(anchorStr) : {};
       dispatch(resetCertificateState());
-      const action = JSON.parse(queryParams.q as string);
+      const action = JSON.parse(query.q as string);
```
Comment (Nebulis) [inline src/components/HomePageContent/HomePageContainer.tsx:45]: what's the type of query.q ? (why need casting)

File: src/pages/newsDetail.tsx [hunk 3]
```diff
@@ -2,16 +2,18 @@ import React, { FunctionComponent } from "react";
...
 import { ResourceDownload } from "./../components/UI/ResourceDownload";
+import { useRouter } from "next/router";
 
 export const NewsPageDetail: FunctionComponent = () => {
-  const location = useLocation();
-  const params: { slug: string } = useParams();
-  const detail: NewsSingle = allNews.find((news) => news.slug === params.slug);
+  const router = useRouter();
+
+  const slug = router.pathname.split("news/")[1];
```
Comment (Nebulis) [inline src/pages/newsDetail.tsx:14]: https://nextjs.org/docs/routing/dynamic-routes

should be retrievable without the need to split?

File: src/AppContainer.tsx (hunk 1 above)
Comment (waynewee) [reply src/AppContainer.tsx:8]: havent refactored fully yet

File: src/pages/newsDetail.tsx (hunk 3 above)
Comment (waynewee) [reply src/pages/newsDetail.tsx:14]: yeap, but i'm trying to refactor bit by bit so this is to allow me to remove react router dom. Will get the actual slug when i actually move the page.

File: src/components/HomePageContent/HomePageContainer.tsx (hunk 2 above)
Comment (waynewee) [reply src/components/HomePageContent/HomePageContainer.tsx:45]: query.q is string | string[] | undefined. type error will be thrown if its not casted

**Pull request merged**